packages/compartment-mapper/src/parse-archive-cjs.js

```diff
@@ -47,6 +51,7 @@ export const parseArchiveCjs = async (
         exports: freeze(exports),
         reexports: freeze(reexports),
         execute: noopExecute,
+        __syncModuleProgram__: syncModuleProgram,
```

Comment:
There's an issue with this change - when it's added here, archive no longer works. 
The presence of `__syncModuleProgram__` causes an error:
`'Property \'__fixedExportMap__\' of a precompiled module record must be an object, got "[undefined]", for module "./index.js"'`

Reply:
Be sure to remove the `execute` method if you’re changing this from being a third-party-static-module-record to a first-party-static-module-record.

Can you contrive a valid `__fixedExportMap__`, `__liveExportMap__`, and any other properties that a 1st-party module source are supposed to have? Are they reasonable defaults? If absent, can the reasonably be inferred from the other properties?

If we’ve got a lot of yes answers, I’m open to moving default beahvior into `ses/src/module-instance.js`.

Reply:
I went with a better isPrecompiled duck typing option https://github.com/endojs/endo/pull/1383/commits/b1cc4c5bf36d9f04438c1de0df30f126c4d2476f#diff-4209ab82d2d256122a0e77a725980708c388be37712ffcd296561976778daa09R65

How do you like it?

I don't think we can come up with correct `__fixedExportMap__` and `__liveExportMap__` for a cjs module without executing it. If we could, we'd have done it instead of creating a thirdparty module concept in ses, right?